bench/app.py

```diff
@@ -62,7 +62,9 @@ def get_app(app, git_url, branch=None, bench='.', build_asset_files=True, verbos
 		restart_supervisor_processes(bench=bench)
 
 def new_app(app, bench='.'):
-	app = app.lower().replace(" ", "_")
+	app = app.lower().replace(" ", "_").replace("-", "_")
+	if not re.match("^[a-z_]*$", app):
```

Comment:
This is validation done in frappe isn't it?

We should duplicate the lowercase and replace in frappe. It should've been in there from the start, but for backward compatibility, we will have to keep it here (put a comment about backward compat)